File: code/spark/spark-streaming-basis/src/main/java/com/heibaiying/utils/JedisPoolUtil.java
package com.heibaiying.utils;

import redis.clients.jedis.Jedis;
import redis.clients.jedis.JedisPool;
import redis.clients.jedis.JedisPoolConfig;

public class JedisPoolUtil {

    /* 声明为volatile防止指令重排序 */
    private static volatile JedisPool jedisPool = null;

    private static final String HOST = "localhost";
    private static final int PORT = 6379;


    /* 双重检查锁实现懒汉式单例 */
    public static Jedis getConnection() {
        if (jedisPool == null) {
            synchronized (JedisPoolUtil.class) {
                if (jedisPool == null) {
                    JedisPoolConfig config = new JedisPoolConfig();
                    config.setMaxTotal(30);
                    config.setMaxIdle(10);
                    jedisPool = new JedisPool(config, HOST, PORT);
                }
            }
        }
        return jedisPool.getResource();
    }
}

File: code/spark/spark-streaming-basis/src/main/java/com/heibaiying/utils/JedisPoolUtil.scala
package com.heibaiying.utils

import redis.clients.jedis.{Jedis, JedisPool, JedisPoolConfig}

object JedisPoolUtil {

  /*创建Jedis连接池*/
  val config = new JedisPoolConfig
  config.setMaxTotal(30)
  config.setMaxIdle(10)
  val jedisPool = new JedisPool(config, "localhost", 6379)


  def getConnection: Jedis = {
    jedisPool.getResource
  }

}
